channel/config_source.go

```diff
@@ -6,28 +6,43 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-type ConfigVersion string
+type ConfigVersion interface {
+	// ID returns this version's unique ID (e.g. a git commit SHA)
+	ID() string
+
+	// Get returns a Config related to the specified version from the local copy.
```

Comment:
No, each `Config` is a self-contained clone (that you're supposed to `Delete()` later). `CachedConfig` just caches `ConfigVersions` to avoid unnecessary git invocations.